independent-projects/arc/runtime/src/main/java/io/quarkus/arc/EventBean.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.arc;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Event;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.enterprise.util.TypeLiteral;
+
+public class EventBean extends BuiltInBean<Event<?>> {
+
+    public static final Set<Type> EVENT_TYPES = Collections
+            .unmodifiableSet(new HashSet<>(Arrays.asList(Event.class, new TypeLiteral<Event<Object>>() {
```

Comment:
Why the `TypeLiteral` when we're only checking the raw type?

Reply:
Not sure, TBF I went after what Weld had and this was there as well.
It was only later when I realized that's not really used, want me to remove that?